### ISSUE TITLE
Setting RouterMandatory on RouterSocket causes InvalidException

### ISSUE THREAD

**Issue by ijrae** [zeromq/netmq#197]
Setting `ZmqSocketOptions.RouterMandatory` on a `RouterSocket` results in an `InvalidException`, for instance the following code doesn't work:
```C#
using (NetMQContext ctx = NetMQContext.Create())
{
    using (RouterSocket socket = ctx.CreateRouterSocket())
    {
        socket.Options.RouterMandatory = true;
    }
}
```
I believe the problematic code is at `NetMQ\zmq\Patterns\Router.cs` line 153:
```C#
if (option == ZmqSocketOptions.RouterRawSocket)
{
    m_rawSocket = (bool)optval;
    if (m_rawSocket)
    {
        m_options.RecvIdentity = false;
        m_options.RawSocket = true;
    }

    return true;
}
else if ( option == ZmqSocketOptions.RouterRawSocket)
{
    m_mandatory = (bool)optval;
    return true;
}
```
The second `else if` should check for `ZmqSocketOptions.RouterMandatory`, not `RouterRawSocket`.

**Comment by ijrae** [comment]
Ah, nevermind, I see a fix for this is in pull request #192.

**Comment by MarcAzar** [comment]
Issue is occuring with newest NETMQ 4.0.1.13
